docs/design/2022-11-17-ttl-table.md

```diff
@@ -0,0 +1,357 @@
+
+```sql
+DELETE FROM t
+WHERE id in (1, 2, 3, ...) AND create_time < '2022-01-01 00:00:00';
+```
+
+Notice that we are still using the condition `create_time < '2022-01-01 00:00:00'` to avoid deleting some "not expired" rows by mistake. For example, a row is expired when scanning, but updated to a not expired value before deleting it.
+
+If there is no secondary index in a TTL table, we can assume that most of the delete operations can do a commit with 1PC. That is because the incoming rows are in the same region in most cases. So if you have a big table, keep no secondary indexes in this table may achieve a better performance for TTL jobs.
+
+#### Time Zone Consideration
+
+Currently, we support three field types as the time column for TTL: `Date`, `DateTime` and `TimeStamp` . However, for `Date` and `DateTime`, they are not an absolute time, and we need another "time zone" information to determine what their accurate time point is. We have two options to select the time zone:
+
+1. Use the system time zone which can be fetched from global variable `system_time_zone` and is determined when the cluster bootstraps. Because the `system_time_zone` will never changes, the row will be deleted at a certain time. However, it will have some problem when the user sets the cluster time zone `time_zone` to a different value other than `SYSTEM`. For example, if the `system_time_zone` is `+08:00` and `time_zone` is set to `UTC`, the record will be seen as expired immediately when the `tidb_ttl_job_run_interval` is less than 8 hours.
+
+2. Use the cluster time zone `time_zone`. It is fine when cluster time zone `time_zone` does not change. But if user changes the `time_zone` settings, the TTL job should be aware of it in time to forbid deleting some unexpected rows. In this case, whether one row is expired or not can not be determined without knowing the `time_zone` at that time.
+
+### New System Variables
+
+Some new system variables will be introduced:
+
+- `tidb_ttl_job_enable`
+  - When this variable is `OFF`, the cluster will stop to schedule TTL jobs and the running jobs will be cancelled.
+  - Scope: Global
+  - Values: [ON, OFF]
+  - Default: ON
+
+- `tidb_ttl_job_run_interval`
+  - The schedule interval between two jobs for one TTL table
+  - Scope: Global
+  - Range: [10m0s, 8760h0m0s]
+  - Default: 1h
+
+- `tidb_ttl_job_schedule_window_start_time`
+  - This variable is used to restrict the start time of the time window of scheduling the ttl jobs. 
+  - Scope: Global
+  - Type: Time
+  - Default: 00:00 +0000
+
+- `tidb_ttl_job_schedule_window_end_time`
+  - This variable is used to restrict the end time of the time window of scheduling the ttl jobs.
+  - Scope: Global
+  - Type: Time
+  - Default: 23:59 +0000
+
+- `tidb_ttl_scan_worker_count`
+  - The count of the scan workers in each TiDB node
+  - Scope: Global
+  - Range: [1, 1024]
```

Comment:
Small nit, there is a const for maxUserConfiguableConcurrency(?) which should apply here. It is currently at 256:
```suggestion
  - Range: [1, 256]
```